File: backend/Models/Event.cs
using System.ComponentModel.DataAnnotations;

namespace backend.Models;

public class Event
{
    [Key]
    public Guid Id { get; set; }
    [Required]
    public Guid WorkspaceId { get; set; }
    [Required]
    public string Title { get; set; } = string.Empty;
    public string? Description { get; set; }
    public string Category { get; set; } = "event";
    public string? Location { get; set; }
    public Guid? CustomerId { get; set; }
    public Customer? Customer { get; set; }
    public ICollection<Employee> AssignedTo { get; set; } = [];
    [Required]
    public DateTime StartDateTime { get; set; }
    [Required]
    public DateTime EndDateTime { get; set; }
    public bool IsAllDay { get; set; } = false;
    public bool IsRecurring { get; set; } = false;
    public Guid? RecurrenceRuleId { get; set; }
    public RecurrenceRule? RecurrenceRule { get; set; }
    [Required]
    public Guid CreatedBy { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
    public DateTime UpdatedAt { get; set; } = DateTime.UtcNow;
}

File: frontend/src/types/Event.ts
import { TEmployee } from './Employee';
import { TAddRecurrenceRule, TRecurrenceRule } from './RecurrenceRule';

export type TEvent = {
  id: string;
  workspaceId: string;
  title: string;
  description?: string;
  category: string;
  location?: string;
  customerId?: string;
  // assignedToIds: string[];
  assignedTo: TEmployee[];
  startDateTime: string;
  endDateTime: string;
  isAllDay: boolean;
  isRecurring: boolean;
  recurrenceRuleId?: string;
  recurrenceRule?: TRecurrenceRule;
  createdBy: string;
  createdAt: string;
  updatedAt: string;
};

export type TAddEvent = {
  workspaceId: string;
  title: string;
  description?: string;
  category: string;
  location?: string;
  customerId?: string;
  assignedToIds: string[];
  startDateTime: string;
  endDateTime: string;
  isAllDay: boolean;
  isRecurring: boolean;
  recurrenceRuleId: string | null;
  recurrenceRule?: TAddRecurrenceRule;
  createdBy?: string;
};

export type TUpdateEvent = {
  id: string;
  workspaceId: string;
  title: string;
  description?: string;
  category: string;
  location?: string;
  customerId?: string;
  assignedToIds: string[];
  startDateTime: string;
  endDateTime: string;
  isAllDay: boolean;
  isRecurring: boolean;
  recurrenceRuleId?: string;
  recurrenceRule?: TAddRecurrenceRule;
};
